packages/eslint-plugin/tests/rules/no-unsafe-return.test.ts

```diff
@@ -127,6 +127,7 @@ function foo(): Set<number> {
         return [] as any[];
       }
     `,
+    'const foo: (() => void) | undefined = () => 1;',
```

Comment:
`functionType.getCallSignatures()` can return an empty array, but that code path was not being tested. This ensures that the code path is now covered.

Reply:
This was brought to light because when I initially refactored the rule to assume that `getCallSignatures()[0]` was always defined, the tests passed, but the rule crashed when linting this repo, therefore showing that a test was missing.